src/chainstate/burn/operations/mod.rs

```diff
@@ -136,6 +145,34 @@ impl From<db_error> for Error {
     }
 }
 
+#[derive(Debug, PartialEq, Clone, Eq, Serialize, Deserialize)]
+pub struct StackStxOp {
+    pub sender: StacksAddress,
+    /// the PoX reward address
+    pub reward_addr: StacksAddress,
+    /// how many ustx this transaction locks
+    pub stacked_ustx: u128,
+    pub num_cycles: u8,
+
+    // common to all transactions
+    pub txid: Txid,                            // transaction ID
+    pub vtxindex: u32,                         // index in the block where this tx occurs
+    pub block_height: u64,                     // block height at which this tx occurs
+    pub burn_header_hash: BurnchainHeaderHash, // hash of the burn chain block header
+}
+
+#[derive(Debug, PartialEq, Clone, Eq, Serialize, Deserialize)]
+pub struct PreStackStxOp {
+    /// the output address
+    pub output: StacksAddress,
+
+    // common to all transactions
+    pub txid: Txid,                            // transaction ID
+    pub vtxindex: u32,                         // index in the block where this tx occurs
+    pub block_height: u64,                     // block height at which this tx occurs
+    pub burn_header_hash: BurnchainHeaderHash, // hash of the burn chain block header
+}
+
```

Comment:
I'm now noticing that neither of these operations contain a consensus hash, even though they have space for it.  Should they contain a consensus hash, so a client can guarantee that these operations will only apply in a node that follows the consensus rules and is on the right PoX fork?

Reply:
I'm not sure what it would add to the user experience. I'd almost certainly prefer that my transaction was picked up regardless of the consensus hash (definitely in the case of the PreStackStx transaction, and probably in the StackStx transaction as well).

Reply:
Fair enough.  The consensus hash's original purpose was to enable the network to survive hard forks -- a consensus hash represents one set of nodes.  But there are other ways to do this, such as by simply changing the magic bytes in each Bitcoin transaction.